packages/web/src/pages/sign-up-page/pages/SelectArtistsPage.tsx

```diff
@@ -64,18 +69,24 @@ export const SelectArtistsPage = () => {
     setCurrentGenre(e.target.value)
   }, [])
 
+  const accountCreationStatus = useSelector(getStatus)
+
   const handleSubmit = useCallback(
     (values: SelectArtistsValues) => {
       const { selectedArtists } = values
       dispatch(addFollowArtists([...selectedArtists]))
       if (isMobile) {
-        navigate(TRENDING_PAGE)
-        setIsWelcomeModalOpen(true)
+        if (accountCreationStatus === EditingStatus.SUCCESS) {
```

Comment:
nice. it is a bit annoying we have this in three spots... could maybe put this in the router logic? this is totally fine though.